src/symbols.c

```diff
@@ -1686,8 +1704,13 @@ static gboolean goto_tag(const gchar *name, gboolean definition)
 }
 
 
-gboolean symbols_goto_tag(const gchar *name, gboolean definition)
+gboolean symbols_goto_tag(const gchar *name, gint pos, gboolean definition)
 {
+	GeanyDocument *doc = document_get_current();
+
+	if (plugin_extension_goto_provided(doc, NULL))
+		return plugin_extension_goto_perform(doc, pos, definition);
```

Comment:
there's still the question of whether this should really be guarded behind Geany thinking there's a word.  IMO it shouldn't, and callers should stop being guarded behind it, and the `perform()` call should return `FALSE` if there's no word to act on.  This would alleviate the artificial dependency on Geany dictating there's a word -- but not actually providing it.

So the callers would look like
```c
static void goto_tag(GeanyDocument *doc, gboolean definition)
{
	if (! symbols_goto_tag(sci_get_current_position(doc->editor->sci), definition))
		utils_beep();
}
```
and
```c
			if (! symbols_goto_tag(editor_info.click_pos, TRUE))
				keybindings_send_command(GEANY_KEY_GROUP_GOTO, GEANY_KEYS_GOTO_MATCHINGBRACE);
			return TRUE;
```

And this here just removes the `name` argument and replaces it with
```c
	const gchar *name;

	/* … */

	editor_find_current_word(doc->editor, pos,
		editor_info.current_word, GEANY_MAX_WORD_LENGTH, NULL);
	name = *editor_info.current_word ? editor_info.current_word : NULL;

	if (! name)
		return FALSE;
```
(and possibly return `TRUE` below when there was a match but it wasn't handled -- or keep it returning `FALSE`, the only thing that happens then is calling *goto mathcing brace*, which might be fine).

---
This might or might not be a detail, it depends on whether extensions depend on the fact there's a word, or if they are written carefully enough not to actually depend on it.

Reply:
> there's still the question of whether this should really be guarded behind Geany thinking there's a word. IMO it shouldn't, and callers should stop being guarded behind it, and the perform() call should return FALSE if there's no word to act on. This would alleviate the artificial dependency on Geany dictating there's a word -- but not actually providing it.

Completely agree - it should be the plugin's decision regardless of what Geany itself considers to be the necessary condition for the goto to be performed. I'll modify the code accordingly.

Reply:
I'd just point out that this eliminates `get_current_word_or_sel()` from the original `goto_tag()` so `editor_find_current_word()` is used both in the ctrl+click and keybinding case. Is the goto selection something worth preserving? I've never used it myself and didn't even know it existed.

Reply:
Ah, right, it calls `get_current_word_or_sel()`, and is indeed different if there's a selection.  I really don't see why we handle selection in this instance, but maybe there's a use case?  I'll try and think about it, but it seems odd (unless it's useful for some specific languages where we might get wordchars wrong?).

Reply:
…or just use `get_current_word_or_sel()`?  Not tested, but the click handler (that doesn't handle the selection) should actually be *clearing* the selection by setting the caret+anchor position, so the selection will always be empty in that code path, won't it?

Reply:
Just tested and yes, I get identical numbers from
```C
int start = sci_get_selection_start(doc->editor->sci);
int end = sci_get_selection_end(doc->editor->sci);
```
placed inside `symbols_goto_tag()` and after control-clicking and having something selected.

Reply:
Just a note, nothing should depend on Geany thinking there is a "word" since its always wrong for languages where non-ASCII is allowed for identifiers.

Reply:
> or just use get_current_word_or_sel()?

But it's static inside `keybindings.c` and we'd need it in `symbols.c`. And with this epochal discovery, I'm going to bed.

Reply:
I'd say 2 or 3 (definitely not 1).  I just pushed a version of 3, given how small the logic is.